cmd/soroban-cli/src/commands/snapshot/create.rs

```diff
@@ -284,29 +309,47 @@ impl Cmd {
                             continue;
                         }
                     };
+
                     if seen.contains(&key) {
                         continue;
                     }
+
                     let keep = match &key {
                         LedgerKey::Account(k) => current.account_ids.contains(&k.account_id),
                         LedgerKey::Trustline(k) => current.account_ids.contains(&k.account_id),
                         LedgerKey::ContractData(k) => current.contract_ids.contains(&k.contract),
                         LedgerKey::ContractCode(e) => current.wasm_hashes.contains(&e.hash),
+                        LedgerKey::ConfigSetting(_) => true,
                         _ => false,
                     };
+
                     if !keep {
                         continue;
                     }
+
                     seen.insert(key.clone());
-                    let Some(val) = val else { continue };
+
+                    let Some(val) = val else {
+                        continue;
+                    };
+
                     match &val.data {
+                        LedgerEntryData::ConfigSetting(ConfigSettingEntry::StateArchival(
+                            state_archival,
+                        )) => {
+                            snapshot.min_persistent_entry_ttl = state_archival.min_persistent_ttl;
+                            snapshot.min_temp_entry_ttl = state_archival.min_temporary_ttl;
+                            snapshot.max_entry_ttl = state_archival.max_entry_ttl;
+                            false
+                        }
+
                         LedgerEntryData::ContractData(e) => {
                             // If a contract instance references contract
                             // executable stored in another ledger entry, add
                             // that ledger entry to the filter so that Wasm for
                             // any filtered contract is collected too in the
                             // second pass.
-                            if keep && e.key == ScVal::LedgerKeyContractInstance {
```

Comment:
`keep` was always `true` at this point (there's a guard to continue the loop if it's false on line 326).